packages/core/src/App/Containers/RealAccountSignup/helpers/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Takes validation result of a form, sets warnings on component's state and pass errors down to the form
+ *
+ * @param {object} values - object containing form field values and validations
+ * @return {object} errors
+ */
+function setWarnsFilterErrors(values) {
+    const warnings = {};
+    Object.keys(values).forEach(field => {
+        if (values[field][0] === 'warn') {
+            warnings[field] = values[field][1];
+            delete values[field];
+        }
+    });
+    this.setState({ warnings });
+    const errors = {};
+    Object.keys(values).forEach(error => (errors[error] = values[error][1]));
+    return errors;
```

Comment:
How about doing a little wishful thinking here? 
```js
function splitErrorTypes (values) {
  // ... the actual implementation
  return {errors, warnings};
};
```
Mind the function name change. 
This way, as per above comment, we can keep this function pure and possibly re-use it across the platform.

Reply:
Renamed to `splitValidationResultTypes` and applied tha change to let the concumer decide how to handle the state of warnings.